src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -272,7 +272,22 @@ private void SetupSingleCheck(CheckFactoryContext checkFactoryContext, string pr
                 // Create the wrapper and register to central context
                 wrapper.StartNewProject(projectFullPath, configurations, userEditorConfigs);
                 var wrappedContext = new CheckRegistrationContext(wrapper, _buildCheckCentralContext);
-                check.RegisterActions(wrappedContext);
+                try
+                {
+                    check.RegisterActions(wrappedContext);
+                }
+                catch (Exception e)
+                {
+                    string message = $"The check '{check.FriendlyName}' failed to register actions with the following message: '{e.Message}'";
+                    if (e.InnerException is null)
+                    {
+                        throw new BuildCheckConfigurationException(message);
+                    }
+                    else
+                    {
+                        throw new BuildCheckConfigurationException(message, e.InnerException);
+                    }
```

Comment:
```suggestion
                    throw new BuildCheckConfigurationException(message, e);
```

Let`s not lose the info